### PR TITLE
Manage libraries modal re-design

### PR DESCRIPTION
**Pull request by maddiedierker**
This PR:
- Updates the "manage libraries" modal to match the new design
- Adds the ability to update a library you've already imported from the "manage libraries" modal

## Design
Before / After
Group 60 (2)

### Before
Screen Shot 2020-03-10 at 4 17 58 PM

### After
Screen Shot 2020-03-09 at 1 43 14 PM

## Video
[Video to display new functionality](https://drive.google.com/file/d/1yQh6-WIdbLHJR6cwLDSnbTttS1469LKl/view?usp=sharing)

## Follow-up Work
- Display the author name for imported libraries _if_ that library has not been unpublished by the author (coming in a follow-up PR)
- Wrap the library description over 2 lines and truncate with an ellipsis ([STAR-1007](https://codedotorg.atlassian.net/browse/STAR-1007))

## Links

- [STAR-967](https://codedotorg.atlassian.net/browse/STAR-967)

# Reviewer Checklist:

- [ ] Tests provide adequate coverage
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/src/code-studio/components/libraries/LibraryListItem.jsx [hunk 1]
```diff
@@ -59,47 +96,66 @@ export default class LibraryListItem extends React.Component {
...
+          <Tooltip text={i18n.viewCode()} place="bottom">
+            <a onClick={this.viewCode} style={styles.libraryTitle}>
+              {library.name}
+            </a>
+          </Tooltip>
           {library.userName && (
-            <span style={styles.author}>Author: {library.userName}</span>
+            <div style={[styles.author, styles.overflowEllipsis]}>
+              <InlineMarkdown
+                markdown={i18n.authorName({name: library.userName})}
```
Comment (Hamms) [inline apps/src/code-studio/components/libraries/LibraryListItem.jsx:111]: `InlineMarkdown` is _not_ safe to use with externally-sourced strings.

We should probably either make it safe or rename it to make that clear

**Review by Hamms — CHANGES_REQUESTED** [review]

File: apps/src/code-studio/components/libraries/LibraryListItem.jsx (hunk 1 above)
Comment (Hamms) [reply apps/src/code-studio/components/libraries/LibraryListItem.jsx:111]: Wait, nevermind! I misremembered; this actually is safe. Sorry for the false alarm!

**Review by Hamms — APPROVED** [review]

**Comment by Hamms** [comment]
I don't know how to remove my review :upside_down_face:

File: apps/src/code-studio/components/libraries/LibraryListItem.jsx (hunk 1 above)
Comment (maddiedierker) [reply apps/src/code-studio/components/libraries/LibraryListItem.jsx:111]: lol that's okay! i appreciate you double-checking 😄

File: apps/src/code-studio/components/libraries/LibraryListItem.jsx [hunk 2]
```diff
@@ -59,47 +96,66 @@ export default class LibraryListItem extends React.Component {
...
     let library = this.props.library;
+
     return (
-      <div className="assetRow" style={styles.listItem}>
-        <span style={styles.libraryName}>{library.name}</span>
-        <span style={styles.description}>
-          {library.description}
-          <br />
+      <div style={styles.listItem}>
+        <div style={[{marginRight: 25}, styles.overflowEllipsis]}>
```
Comment (jmkulwik) [inline apps/src/code-studio/components/libraries/LibraryListItem.jsx:102]: Whoah. Didn't realize you could do this!!

File: apps/src/code-studio/components/libraries/LibraryManagerDialog.jsx [hunk 3]
```diff
@@ -109,6 +118,16 @@ export default class LibraryManagerDialog extends React.Component {
     this.setState({libraries: dashboard.project.getProjectLibraries()});
   };
 
+  updateLibraryInProject = libraryJson => {
+    let libraries = [...this.state.libraries];
+    const libIndex = libraries.findIndex(
```
Comment (jmkulwik) [inline apps/src/code-studio/components/libraries/LibraryManagerDialog.jsx]: NIT: rename `libIndex` to `libraryIndex` or just `index`

File: apps/src/code-studio/components/libraries/LibraryManagerDialog.jsx [hunk 4]
```diff
@@ -227,11 +248,11 @@ export default class LibraryManagerDialog extends React.Component {
           style={{...styles.dialog, ...(isViewingCode ? styles.hidden : {})}}
           useUpdatedStyles
         >
-          <div style={styles.header}>{i18n.libraryManage()}</div>
+          <h1 style={styles.header}>{i18n.libraryManage()}</h1>
```
Comment (jmkulwik) [inline apps/src/code-studio/components/libraries/LibraryManagerDialog.jsx:270]: 🎉

File: apps/src/code-studio/components/libraries/LibraryManagerDialog.jsx [hunk 5]
```diff
@@ -43,10 +44,18 @@ const styles = {
...
-    display: 'flex'
+    display: 'flex',
+    alignItems: 'baseline'
   },
   add: {
-    margin: DEFAULT_MARGIN
+    margin: DEFAULT_MARGIN,
+    color: color.dark_charcoal,
+    borderColor: color.dark_charcoal,
+    ':disabled': {
```
Comment (jmkulwik) [inline apps/src/code-studio/components/libraries/LibraryManagerDialog.jsx:54]: NIT (up to you): if you change the add button to a `<Button>`, you shouldn't need the `disabled` styles.

**Review by jmkulwik — APPROVED** [review]
🎉

File: apps/src/code-studio/components/libraries/LibraryListItem.jsx (hunk 2 above)
Comment (maddiedierker) [reply apps/src/code-studio/components/libraries/LibraryListItem.jsx:102]: that's from our `Radium` package! very useful for stuff like this case where you need shared styles

**Pull request merged**